src/DistributionApi.php

```diff
@@ -154,8 +155,15 @@ private function createDownloadDirectoryIfNeeded(string $downloadPath): void
     private function getResponse(string $endpoint): array
     {
         /** @var array<array<string, string>> $response */
-        $response = json_decode($this->circruitBreaker->call($endpoint), true) ?: [];
+        $response = json_decode($this->circruitBreaker->call($this->addShopInfoToAPIEndpoint($endpoint)), true) ?: [];
 
         return $response;
     }
+
+    private function addShopInfoToAPIEndpoint(string $endpoint): string
+    {
+        $shopUrl = urlencode(Tools::getShopDomain());
+
+        return sprintf('%s?shop_domain=%s', $endpoint, $shopUrl);
```

Comment:
```suggestion
        return sprintf('%s?shop_domain=%s', $endpoint, urlencode($shopUrl));
```
Also, what if `$endpoint` already contains a query string?

Reply:
You want `urlencode` to be called twice? 😅
For now, `$endpoint` cannot contain a query string, maybe if one day we change it, we'll see what we should do but for now we can keep it simple?